sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1385,10 +1385,6 @@ def test_convergence_warnings():
     X = random_state.standard_normal((1000, 500))
     y = random_state.standard_normal((1000, 3))
 
-    # check that the model fails to converge (a negative dual gap cannot occur)
-    with pytest.warns(ConvergenceWarning):
-        MultiTaskElasticNet(max_iter=1, tol=-1).fit(X, y)
-
     # check that the model converges w/o convergence warnings
```

Comment:
Rather than deleting this test, we can instead set `tol=0` to achieve the same effect. You'll need to change the comment above to match.

Reply:
I tried setting tol=0 in this part of the test but I seem to get back:
Failed: DID NOT WARN. No warnings of type (<class 'sklearn.exceptions.ConvergenceWarning'>,) were emitted. The list of emitted warnings is: [].

Reply:
Seems it is random-seed dependent. Turns out that the convergence criteria:
https://github.com/scikit-learn/scikit-learn/blob/db30f6c1875cc0ae0f6c178d0f9e7a6fe5eed3cf/sklearn/linear_model/_cd_fast.pyx#L945-L947

can produce false-positives when `tol=0`. Namely `gap` takes on small negative values despite mathematically being bounded below by $0$.

Well this code path is also tested in `test_multi_task_lasso_and_enet` so I think it's fine to remove the test as you initially had. Feel free to resolve this comment, thanks.

Reply:
Thank you for the explanation.